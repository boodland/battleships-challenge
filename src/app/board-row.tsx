import BoardCell from './board-cell';
import { Ship, columns, hasShip } from './utils';

export interface BoardRowProps {
  startPosition: number;
  ships: Ship[];
  shots: boolean[];
  showShips: boolean;
  onCellClick: (index: number) => void;
}

function BoardRow({
  startPosition,
  ships,
  shots,
  showShips,
  onCellClick,
}: BoardRowProps) {
  const boardRowSquares = columns.map(
    (column, index) => index + startPosition * 10
  );
  return (
    <>
      {boardRowSquares.map((position) => (
        <BoardCell
          key={position}
          shotted={shots[position]}
          hasShip={hasShip(ships, position)}
          showShip={showShips}
          onCellClick={() => onCellClick(position)}
        />
      ))}
    </>
  );
}

export default BoardRow;
