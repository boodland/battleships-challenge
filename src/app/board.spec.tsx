import { render } from '@testing-library/react';

import Board from './board';
import { BoardRowProps } from './board-row';

vi.mock('./utils', () => {
  return { rows: [0, 1, 2, 3, 4], columns: ['A', 'B', 'C', 'D', 'E'] };
});
vi.mock('./board-row', () => ({
  default: (props: BoardRowProps) =>
    `MockedBoardRow ${JSON.stringify(props)} onCellClick: ${
      props.onCellClick.name
    }`,
}));

const onShot = (position: number) => null;

describe('Board', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <Board ships={[]} shots={[]} showShips={false} onShot={onShot} />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should display columns row', () => {
    const { container } = render(
      <Board ships={[]} shots={[]} showShips={false} onShot={onShot} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should display one BoardRow per row', () => {
    const { container } = render(
      <Board ships={[]} shots={[]} showShips={false} onShot={onShot} />
    );
    expect(container).toMatchSnapshot();
  });
});
