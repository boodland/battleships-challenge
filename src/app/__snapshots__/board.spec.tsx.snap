// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Board > should display columns row 1`] = `
<div
  class="_columns_672456"
>
  <span
    class="_text_672456"
  >
    A
  </span>
  <span
    class="_text_672456"
  >
    B
  </span>
  <span
    class="_text_672456"
  >
    C
  </span>
  <span
    class="_text_672456"
  >
    D
  </span>
  <span
    class="_text_672456"
  >
    E
  </span>
</div>
`;

exports[`Board > should display one BoardRow per row 1`] = `
<div>
  <div
    class="_columns_672456"
  >
    <span
      class="_text_672456"
    >
      A
    </span>
    <span
      class="_text_672456"
    >
      B
    </span>
    <span
      class="_text_672456"
    >
      C
    </span>
    <span
      class="_text_672456"
    >
      D
    </span>
    <span
      class="_text_672456"
    >
      E
    </span>
  </div>
  <div>
    <span
      class="_text_672456"
    >
      1
    </span>
    MockedBoardRow {"startPosition":0,"ships":[],"shots":[],"showShips":false} onCellClick: onShot
  </div>
  <div>
    <span
      class="_text_672456"
    >
      2
    </span>
    MockedBoardRow {"startPosition":1,"ships":[],"shots":[],"showShips":false} onCellClick: onShot
  </div>
  <div>
    <span
      class="_text_672456"
    >
      3
    </span>
    MockedBoardRow {"startPosition":2,"ships":[],"shots":[],"showShips":false} onCellClick: onShot
  </div>
  <div>
    <span
      class="_text_672456"
    >
      4
    </span>
    MockedBoardRow {"startPosition":3,"ships":[],"shots":[],"showShips":false} onCellClick: onShot
  </div>
  <div>
    <span
      class="_text_672456"
    >
      5
    </span>
    MockedBoardRow {"startPosition":4,"ships":[],"shots":[],"showShips":false} onCellClick: onShot
  </div>
</div>
`;
